### ISSUE TITLE
Change to the structure for the extraction (args)

### ISSUE THREAD

**Issue by BeckySharp** [clulab/tomcat-text#33]
Joe:

We need to change the `arguments` to a `Seq[DialogAgentMessageDataExtraction]` (potentially empty)

https://github.com/clulab/tomcat-text/blob/a4e72afec04dc548abea319e0b226fea85d31b00/src/main/scala/org/clulab/asist/messages/DialogAgentMessage.scala#L40

To get this, we need to modify this:
https://github.com/clulab/tomcat-text/blob/a9851c32ef32194e680c889b6b7999173a099202/src/main/scala/org/clulab/asist/DialogAgent.scala#L51-L53

Basically, rather than flattening the args into a list of their string labels (which you join in line 64), we need to recursively call `extraction()` on them and make a list of `DialogAgentMessageDataExtraction` or of Option[...] (didn't read closely enough to see why it's an option).

I noticed some other things, I'll make a separate issue for them.

**Comment by BeckySharp** [comment]
@jastier oops! I forgot to say **thanks**!!! :)

**Comment by jastier** [comment]
This was completed during the 2021 March Hackathon.
